packages/babel-plugin/src/utils/build-styled-component.ts

```diff
@@ -210,13 +200,73 @@ const styledTemplate = (opts: StyledTemplateOpts, meta: Metadata): t.Node => {
         const nestedArrowFunctionExpressionVisitor = {
           noScope: true,
           MemberExpression(path: NodePath<t.MemberExpression>) {
-            const propsToDestructureFromMemberExpression =
-              handleMemberExpressionInStyledInterpolation(path);
+            const traversedUpFunctionPath: NodePath<t.Node> | null = path.find((parentPath) =>
+              parentPath.isFunction()
+            );
+
+            const memberExpressionKeyName = (path.node.object as t.Identifier).name;
```

Comment:
Why the type assertion here?

Reply:
I thought I could simplify https://github.com/atlassian-labs/compiled/blob/master/packages/babel-plugin/src/utils/build-styled-component.ts#L154-L157 assuming that it is going to be an identifier ( I have never seen other types)

Looking at this again, it seems that [object should be an expression](https://babeljs.io/docs/en/babel-types#memberexpression); I'll keep the initial logic

Reply:
keep initial logic